doc/developer-guides/hld/hld-splitlock.rst

```diff
@@ -0,0 +1,122 @@
+.. _hld_splitlock:
+
+Handling Split-locked Access in Acrn
+####################################
+
+This document explains what is Split-locked Access, how to detect it and what
+Acrn do for handling it.
+
+Split-locked Access Introduction
+********************************
+The Intel-64 and IA32 multiple-processor systems support locked atomic
+operations on locations in system memory. For example, The LOCK instruction
+prefix be prepended to the following instructions: ADD, ADC, AND, BTC, BTR, BTS,
+CMPXCHG, CMPXCH8B, CMPXCHG16B, DEC, INC, NEG, NOT, OR, SBB, SUB, XOR, XADD,
+XCHG. and these instructions must with memory destination operand forms.
+Reading or writing a byte in system memory are always guaranteed to be handled
+atomically, otherwise, these locked atomic operations can impact system in two
+ways:
+
+- The destination operand located in same cache line. Cache coherency protocols
+  that ensure that atomic operations can be carried out on cached data
+  structures with cache lock.
+
+- The destination operand located in two cache line. This atomic operation is
+  called a Split-locked Access. For this situation, LOCK# bus signal is asserted
+  to lock the system bus, to ensure they are atomic. See `Intel 64 and IA-32 Architectures Software Developer's Manual(SDM), Volume 3, (Section 8.1.2 Bus Locking) <https://software.intel.com/en-us/download/intel-64-and-ia-32-architectures-sdm-combined-volumes-3a-3b-3c-and-3d-system-programming-guide>`_.
+
+Split-locked Access may lead to long latency, to ordinary memory operations on
+other CPUs, which is hard to investigate.
+
+Split-locked Access Detection
+*****************************
+From the Microarchitecture Tremont, a new CPU capability is introduced for
```

Comment:
Is there a good (public) link we can give for people who want to know more about that microarchitecture? Ideally a link where they can also easily determine if a particular CPU family is based on it or not? I was thinking of using ark.intel.com but couldn't find an obvious place in there for that.

Reply:
Sorry, I can't find a way to determine what microarchitecture is of a  particular cpu, except the CPU vendor told me that. I usually search the publish news of  the Processor Type, maybe it contains the code name of microarchitecture. I think that name is useful for CPU manufacturer. For other people, it is extra information.  If people want to know the capability of Intel microarchitechures, can find some in https://software.intel.com/sites/default/files/managed/9e/bc/64-ia-32-architectures-optimization-manual.pdf

Reply:
Fair enough, while I like the idea of giving as much info as possible, I'm wondering if a better approach wouldn't be to simply be vague and mention some new processors support this feature. A bit like CAT which is supported on some processors, but not all and instead of listing the ones that do, we just mention how to figure this out programmatically (something you are already describing below).

@dbkinder @deb-intel , your thoughts?

Reply:
Yes, that has details about Termont. Do we need to add the link in to the Doc? It seems this document is about split-locked access. Some people have triggered that exception on EHL & TGL platform, with latest HV, when use OVMF launch VM. They may feel strange why the VM  can't be launched any more. This doc may help them to know what happened, can we merge this first, and I can start to do the next work about this feature, thanks. By the way, I also want to have a update this time about how to use HV shell to disable splitlock detection, to help EHL & TGL Acrn users.